bin/bench-prover/src/bench_functions.rs

```diff
@@ -0,0 +1,102 @@
+use miden_objects::{
+    Felt,
+    account::Account,
+    asset::{Asset, AssetVault, FungibleAsset},
+    note::NoteType,
+    testing::account_id::ACCOUNT_ID_SENDER,
+};
+use miden_testing::{Auth, MockChain, utils::prove_and_verify_transaction};
+
+pub fn prove_consume_note_with_new_account() {
```

Comment:
This currently includes non-trivial setup of the function (e.g. `MockChain::new` or `    mock_chain.prove_next_block();`) in the benchmark. But I think the goal of the benchmark should be to measure only the proving time (also excluding the verification), so really only the call to `LocalTransactionProver::prove`, or am I missing the point? Right now, the code is basically doing the same as one of our integration tests, but the goal here should be a different one imo.

So the question is, can we change the benchmark such that this setup happens outside the measurements (for both of the benchmarks)?

Reply:
I changed the benchmarks to only test the proving part. I included the serialization of the proof as well as part of the measurement as well. Happy to make changes again here, we can modify exactly what is being measured in the benchmark.